###### Reading Test Scores

In [2]:
pisa_train=read.csv("Data Files/pisa2009train.csv")

In [3]:
pisa_test=read.csv("Data Files/pisa2009test.csv")

In [4]:
summary(pisa_train)

     grade            male                      raceeth       preschool     
 Min.   : 8.00   Min.   :0.0000   White             :2015   Min.   :0.0000  
 1st Qu.:10.00   1st Qu.:0.0000   Hispanic          : 834   1st Qu.:0.0000  
 Median :10.00   Median :1.0000   Black             : 444   Median :1.0000  
 Mean   :10.09   Mean   :0.5111   Asian             : 143   Mean   :0.7228  
 3rd Qu.:10.00   3rd Qu.:1.0000   More than one race: 124   3rd Qu.:1.0000  
 Max.   :12.00   Max.   :1.0000   (Other)           :  68   Max.   :1.0000  
                                  NA's              :  35   NA's   :56      
 expectBachelors     motherHS    motherBachelors    motherWork    
 Min.   :0.0000   Min.   :0.00   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:1.0000   1st Qu.:1.00   1st Qu.:0.0000   1st Qu.:0.0000  
 Median :1.0000   Median :1.00   Median :0.0000   Median :1.0000  
 Mean   :0.7859   Mean   :0.88   Mean   :0.3481   Mean   :0.7345  
 3rd Qu.:1.0000   3rd Qu.:1.00   3rd Qu.:1.0000  

In [5]:
(tapply(pisa_train$readingScore,pisa_train$male,mean))

0        1 
512.9406 483.5325

    1 gives Male mean reading score whereas 0 gives Female reading score

In [6]:
pisa_train = na.omit(pisa_train)

pisa_test = na.omit(pisa_test)

    Removes NA fields

In [7]:
summary(pisa_train)

     grade            male                                          raceeth    
 Min.   : 8.00   Min.   :0.0000   American Indian/Alaska Native         :  20  
 1st Qu.:10.00   1st Qu.:0.0000   Asian                                 :  95  
 Median :10.00   Median :1.0000   Black                                 : 228  
 Mean   :10.13   Mean   :0.5012   Hispanic                              : 500  
 3rd Qu.:10.00   3rd Qu.:1.0000   More than one race                    :  81  
 Max.   :12.00   Max.   :1.0000   Native Hawaiian/Other Pacific Islander:  20  
                                  White                                 :1470  
   preschool      expectBachelors     motherHS     motherBachelors 
 Min.   :0.0000   Min.   :0.0000   Min.   :0.000   Min.   :0.0000  
 1st Qu.:0.0000   1st Qu.:1.0000   1st Qu.:1.000   1st Qu.:0.0000  
 Median :1.0000   Median :1.0000   Median :1.000   Median :0.0000  
 Mean   :0.7274   Mean   :0.8343   Mean   :0.896   Mean   :0.3637  
 3rd Qu.:1.0000   3r

In [59]:
str(pisa_train)

'data.frame':	2414 obs. of  24 variables:
 $ grade                : int  11 10 10 10 10 10 10 10 11 9 ...
 $ male                 : int  1 0 1 0 1 0 0 0 1 1 ...
 $ raceeth              : Factor w/ 7 levels "White","American Indian/Alaska Native",..: 1 4 5 1 6 5 1 5 1 1 ...
 $ preschool            : int  0 1 1 1 1 1 1 1 1 1 ...
 $ expectBachelors      : int  0 1 0 1 1 1 1 0 1 1 ...
 $ motherHS             : int  1 0 1 1 1 1 1 0 1 1 ...
 $ motherBachelors      : int  1 0 0 0 1 0 0 0 0 1 ...
 $ motherWork           : int  1 1 1 0 1 1 1 0 0 1 ...
 $ fatherHS             : int  1 1 1 1 0 1 1 0 1 1 ...
 $ fatherBachelors      : int  0 0 0 0 0 0 1 0 1 1 ...
 $ fatherWork           : int  1 1 0 1 1 0 1 1 1 1 ...
 $ selfBornUS           : int  1 1 1 1 1 0 1 0 1 1 ...
 $ motherBornUS         : int  1 1 1 1 1 0 1 0 1 1 ...
 $ fatherBornUS         : int  1 1 0 1 1 0 1 0 1 1 ...
 $ englishAtHome        : int  1 1 1 1 1 0 1 0 1 1 ...
 $ computerForSchoolwork: int  1 1 1 1 1 0 1 1 1 1 ...
 $ read30Mi

In [10]:
pisa_train$raceeth = relevel(pisa_train$raceeth, "White")

In [11]:
pisa_test$raceeth = relevel(pisa_test$raceeth, "White")

In [12]:
lmScore=lm(readingScore ~ .,data=pisa_train)

    _lm(DependentVar ~.,data=DataVar) . indicates that all independent variables are used to create the model, saves the hassle of typing them out_

In [14]:
summary(lmScore)


Call:
lm(formula = readingScore ~ ., data = pisa_train)

Residuals:
    Min      1Q  Median      3Q     Max 
-247.44  -48.86    1.86   49.77  217.18 

Coefficients:
                                                Estimate Std. Error t value
(Intercept)                                   143.766333  33.841226   4.248
grade                                          29.542707   2.937399  10.057
male                                          -14.521653   3.155926  -4.601
raceethAmerican Indian/Alaska Native          -67.277327  16.786935  -4.008
raceethAsian                                   -4.110325   9.220071  -0.446
raceethBlack                                  -67.012347   5.460883 -12.271
raceethHispanic                               -38.975486   5.177743  -7.528
raceethMore than one race                     -16.922522   8.496268  -1.992
raceethNative Hawaiian/Other Pacific Islander  -5.101601  17.005696  -0.300
preschool                                      -4.463670   3.486055  -1.28

    Low multiple R2 does not necessarily imply that the model is of poor quality. More often than not, it simply means that the prediction problem at hand is complex

In [20]:
SSE=sum(lmScore$residuals^2)

In [26]:
SST=sum((mean(pisa_train$readingScore)-lmScore$residuals)^2)

In [33]:
RMSE=sqrt(SSE/nrow(pisa_train))

In [34]:
SSE

[1] 12993365

In [35]:
SST

[1] 660634689

In [36]:
RMSE

[1] 73.36555

In [49]:
RMSE_alternate=sqrt(mean(lmScore$residuals^2))

    _The coefficient 29.54 on grade is the difference in reading score between two students who are identical other than having a difference in grade of 1. Because A and B have a difference in grade of 2, the model predicts that student A has a reading score that is 2*29.54 larger._

In [50]:


RMSE_alternate

[1] 73.36555

In [51]:
predictlm=predict(lmScore,newdata=pisa_test)

    What is the sum of squared errors (SSE) of lmScore on the testing set?

In [60]:
SSE=sum((pisa_test$readingScore-predictlm)^2)

In [61]:
SSE

[1] 5762082

In [56]:
max(predictlm)-min(predictlm)

[1] 284.4683

    What is the root-mean squared error (RMSE) of lmScore on the testing set?

In [62]:
sqrt(SSE/nrow(pisa_test))

[1] 76.29079

    What is the predicted test score used in the baseline model? Remember to compute this value using the training set and not the test set.

In [63]:
mean(pisa_train$readingScore)

[1] 517.9629

    What is the sum of squared errors of the baseline model on the testing set? HINT: We call the sum of squared errors for the baseline model the total sum of squares (SST).

In [66]:
SST=sum((mean(pisa_train$readingScore)-pisa_test$readingScore)^2)

In [67]:
SST

[1] 7802354

In [71]:
R2=1-SSE/SST

In [72]:
R2

[1] 0.2614944